docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,109 @@
+# Add generic fetcher
+
+- Status: proposed
+- Date: 2024-10-30
+
```

Comment:
@brunoapimentel did we want ^this to be captured? I think in your ADR template attempt we said this bit of information isn't really useful to anyone, or do I remember this wrong?

Reply:
Removed.